### ISSUE TITLE
analog read exception

### ISSUE THREAD

**Issue by 4A5G0027** [dlyckelid/IOExpander-TLA2518#1]
For the lib used, only analog 0 will affect the values of all pins, and analog 1-7 will not change. How much does it take to read out SYSTEM_STATUS?

**Comment by dlyckelid** [comment]
Hi!

That is strange but I havent really tested analog reading since before I wrote the library. I just had it in my code before and then it worked but will test it with my hardware and see if I get the same issue. I have analog sources connected to input 2-4 on my hardware so cannot test all of them but it should show if i have the same issue as you have.

Reading systemstatus should not be to hard but perhaps simplify it into different methods to read the different things that the Systemstatus returns?

**Comment by 4A5G0027** [comment]
Hi

Check out the IOExpander-TLA2518 lib
Analog pin setting, only modify 2bit is OK?
Are there any other modifications

![1673944734973](https://user-images.githubusercontent.com/48422997/212849818-fdb65bf5-caeb-48eb-a943-3047c5daa260.jpg)

**Comment by dlyckelid** [comment]
Yes that should be alright. Also the chip is default to analog input setting so could you try to powercycle it and try without doing any pin-settings?

**Comment by 4A5G0027** [comment]
How to confirm the process of powercycle with the program?

Can you provide a method?

**Comment by dlyckelid** [comment]
I have done a new release now with a reset() method. I tested with my hardware and I can read the different analog inputs without issues with the last release.
I also made all other methods public so now you can get the systemstatus by calling
`ioexpander->readRegister(SYSTEM_STATUS_ADDRESS)`

**Comment by mislamhoriba** [comment]
i am new to Arduino. I have tried to run your example code and getting the error below. Could you please update the example?
TLA2518Test:17: error: 'VSPI' was not declared in this scope
 SPIClass vspi(VSPI);
               ^
TLA2518Test: In function 'void setup()':
TLA2518Test:29: error: 'ANALOG' was not declared in this scope
       ioexpander->pinModeIO(2,ANALOG);

**Comment by dlyckelid** [comment]
> i am new to Arduino. I have tried to run your example code and getting the error below. Could you please update the example? TLA2518Test:17: error: 'VSPI' was not declared in this scope SPIClass vspi(VSPI); ^ TLA2518Test: In function 'void setup()': TLA2518Test:29: error: 'ANALOG' was not declared in this scope ioexpander->pinModeIO(2,ANALOG);

What type of arduino device are you using?
The example is made for ESP32 device so some changes might be needed in able to make it work on other devices. For me it compiles and works as is.

**Comment by mislamhoriba** [comment]
I am using teensy 4.1
to read analog input--------
teensy pin 39 to TLA2518 pin SDO
teensy pin 40 to TLA2518 pin CS
teensy pin 26 to TLA2518 pin SDI
teensy pin 27 to TLA2518 pin SCLK
